src/main/java/de/dennisguse/opentracks/settings/PreferencesOpenHelper.java

```diff
@@ -1,34 +1,65 @@
 package de.dennisguse.opentracks.settings;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
 import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.util.CsvConstants;
 
 class PreferencesOpenHelper {
 
-    private static final int PREFERENCES_VERSION = 1;
+    private final int version;
 
-    private PreferencesOpenHelper() {
+    private PreferencesOpenHelper(int version) {
+        this.version = version;
     }
 
-    static PreferencesOpenHelper newInstance() {
-        return new PreferencesOpenHelper();
+    static PreferencesOpenHelper newInstance(int version) {
+        return new PreferencesOpenHelper(version);
     }
 
-    void checkForUpgrade() {
+    void check() {
         int lastVersion = PreferencesUtils.getInt(R.string.prefs_last_version_key, 0);
-        if (PREFERENCES_VERSION > lastVersion) {
+        if (version > lastVersion) {
             onUpgrade();
+        } else if (version < lastVersion) {
+            onDowngrade();
         }
     }
 
     private void onUpgrade() {
-        PreferencesUtils.setInt(R.string.prefs_last_version_key, PREFERENCES_VERSION);
-        switch (PREFERENCES_VERSION) {
-            case 1:
-                upgradeFrom0to1();
+        PreferencesUtils.setInt(R.string.prefs_last_version_key, version);
+        for (int i = 1; i <= version; i++) {
+            switch (i) {
+                case 1:
+                    upgradeFrom0to1();
+                    break;
+                case 2:
+                    upgradeFrom1to2();
+                    break;
+            }
```

Comment:
add default case with throw new RuntimeException: somebody like me might forget to add the new cases here :)